File: 프로그래머스/1/161989. 덧칠하기/덧칠하기.js
function solution(n, m, section) {
    var answer = 0;
    let index = 0;
    let last_paint = -1;
    
    for (let tile = 1; tile <= n; tile++) {
        if (index == section.length || section[index] != tile) continue;
        if (last_paint == -1 || last_paint + m <= tile) {
            answer += 1
            last_paint = tile
        }
        index += 1
    }
    
    return answer;
}

File: 프로그래머스/1/161989. 덧칠하기/덧칠하기.py
def solution(n, m, section):
    answer = 0
    
    last_paint = -1
    index = 0
    
    for tile in range(1, n+1):
        if index == len(section) or tile != section[index]: continue
        if last_paint == -1 or last_paint + m <= tile:
            answer += 1
            last_paint = tile
        index += 1

    return answer